src/OrchardCore.Modules/OrchardCore.PublishLater/Migrations.cs

```diff
@@ -1,59 +1,84 @@
-
-            SchemaBuilder.CreateMapIndexTable<PublishLaterPartIndex>(table => table
-                .Column<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
-
-            SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
-                .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
-                    "DocumentId",
-                    nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
-
-            // Shortcut other migration steps on new content definition schemas.
-            return 2;
-        }
-
-        // This code can be removed in a later version.
-        public int UpdateFrom1()
-        {
-            // The 'ScheduledPublishUtc' column and related index are kept on existing databases,
-            // this because dropping an index and altering a column don't work on all providers.
+        SchemaBuilder.CreateMapIndexTable<PublishLaterPartIndex>(table => table
+            .Column<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
+            .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
+                "DocumentId",
+                nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        // Shortcut other migration steps on new content definition schemas.
+        return 2;
+    }
 
-            SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
-                .AddColumn<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
+    // This code can be removed in a later version.
+    public int UpdateFrom1()
+    {
+        // The 'ScheduledPublishUtc' column and related index are kept on existing databases,
+        // this because dropping an index and altering a column don't work on all providers.
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
+            .AddColumn<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
+            .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
+                "DocumentId",
+                nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        return 2;
+    }
+
+    public int UpdateFrom2()
+    {
+        SchemaBuilder.CreateMapIndexTable<PublishLaterPartIndex>(table => table
+            .Column<string>(nameof(PublishLaterPartIndex.ContentItemId))
+            .Column<bool>(nameof(PublishLaterPartIndex.Latest))
+            .Column<bool>(nameof(PublishLaterPartIndex.Published))
+            .Column<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
```

Comment:
Here you should not create an index but only add the columns that were missings, e.g. Published and Latest at the end, something like this for each column.

        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
            .AddColumn<bool>("xxx")
        );

Reply:
You are right, I was looking for that at the time, seems I typed a wrong key stroke ;)

Reply:
Seems I was looking for `AlterMapIndexTable` ;)

Reply:
See my other comments for more details